app/Services/InboundMail/InboundMailEngine.php

```diff
@@ -0,0 +1,279 @@
+<?php
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2023. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+namespace App\Services\InboundMail;
+
+use App\Events\Expense\ExpenseWasCreated;
+use App\Factory\ExpenseFactory;
+use App\Jobs\Util\SystemLogger;
+use App\Libraries\MultiDB;
+use App\Models\ClientContact;
+use App\Models\Company;
+use App\Models\SystemLog;
+use App\Models\Vendor;
+use App\Models\VendorContact;
+use App\Services\InboundMail\InboundMail;
+use App\Utils\Ninja;
+use App\Utils\TempFile;
+use App\Utils\Traits\GeneratesCounter;
+use App\Utils\Traits\SavesDocuments;
+use App\Utils\Traits\MakesHash;
+use Cache;
+use Illuminate\Queue\SerializesModels;
+use Log;
+
+class InboundMailEngine
+{
+    use SerializesModels, MakesHash;
+    use GeneratesCounter, SavesDocuments;
+
+    private ?Company $company;
```

Comment:
Do we need global ref to $company?

Reply:
this was a leftover of the old implementation. i removed it, thanks :)